tests/agent_server/conftest.py

```diff
@@ -441,6 +449,15 @@ class BadPost(Provider):
         def post(self, ctx, resource) -> None:
             raise Exception()
 
+    class Empty:
+        pass
+
+    @provider("test::BadLogging", name="test_bad_logging")
+    class BadLogging(Provider):
+        def do_changes(self, ctx, resource, changes):
+            ctx.info("This is not JSON serializable: %(val)s", val=Empty())
+            super().do_changes(ctx, resource, changes)
```

Comment:
This line is redundant. The super class is an abstract method that raises a `NotImplementedError`.